fe/fe-core/src/main/java/org/apache/doris/common/util/S3Util.java

```diff
@@ -39,32 +40,114 @@
+            boolean isUsePathStyle) {
+        EqualJitterBackoffStrategy backoffStrategy = EqualJitterBackoffStrategy
+                .builder()
+                .baseDelay(Duration.ofSeconds(1))
+                .maxBackoffTime(Duration.ofMinutes(1))
+                .build();
+        // retry 3 time with Equal backoff
+        RetryPolicy retryPolicy = RetryPolicy
+                .builder()
+                .numRetries(3)
+                .backoffStrategy(backoffStrategy)
+                .build();
+        ClientOverrideConfiguration clientConf = ClientOverrideConfiguration
+                .builder()
+                // set retry policy
+                .retryPolicy(retryPolicy)
+                // using AwsS3V4Signer
+                .putAdvancedOption(SdkAdvancedClientOption.SIGNER, AwsS3V4Signer.create())
+                .build();
+        return S3Client.builder()
+                .httpClient(UrlConnectionHttpClient.create())
+                .endpointOverride(endpoint)
+                .credentialsProvider(getAwsCredencialsProvider(credential))
+                .region(Region.of(region))
+                .overrideConfiguration(clientConf)
+                // disable chunkedEncoding because of bos not supported
+                .serviceConfiguration(S3Configuration.builder()
+                        .chunkedEncodingEnabled(false)
+                        .pathStyleAccessEnabled(isUsePathStyle)
+                        .build())
+                .build();
+    }
+
+    /**
+     * creating different credentials provider when creating s3client
+     * @param endpoint
+     * @param region
+     * @param accessKey
+     * @param secretKey
+     * @param sessionToken
+     * @param roleArn
+     * @param externalId
+     * @return
+     */
+    private static AwsCredentialsProvider getAwsCredencialsProvider(URI endpoint, String region, String accessKey,
+            String secretKey, String sessionToken, String roleArn, String externalId) {
+
+        if (!Strings.isNullOrEmpty(accessKey) && !Strings.isNullOrEmpty(secretKey)) {
+            if (Strings.isNullOrEmpty(sessionToken)) {
+                return StaticCredentialsProvider.create(AwsBasicCredentials.create(accessKey, secretKey));
+            } else {
+                return StaticCredentialsProvider.create(AwsSessionCredentials.create(accessKey,
+                        secretKey, sessionToken));
+            }
+        }
+
+        if (!Strings.isNullOrEmpty(roleArn)) {
+            StsClient stsClient = StsClient.builder()
+                    .credentialsProvider(InstanceProfileCredentialsProvider.create())
+                    .build();
+            return StsAssumeRoleCredentialsProvider.builder()
+                    .stsClient(stsClient)
+                    .refreshRequest(r -> r.roleArn(roleArn).externalId(externalId)
+                            .roleSessionName("aws-sdk-java-v2-fe"))
+                    .build();
```

Comment:
When a user configures AK/SK/ARN and other parameters simultaneously (or perhaps we should limit input to only one option?), the actual authentication method differs from the default AWS behavior, and the behavior seems different between the BE and FE. It looks like the ARN is actually used in the BE, while the FE uses AK/SK.

Reply:
> When a user configures AK/SK/ARN and other parameters simultaneously (or perhaps we should limit input to only one option?), the actual authentication method differs from the default AWS behavior, and the behavior seems different between the BE and FE. It looks like the ARN is actually used in the BE, while the FE uses AK/SK.

fe/be both have the same logical, if ak/sk are not empty, it will prefer to use ak/sk